lib/saluki-components/src/destinations/datadog_status_flare/mod.rs

```diff
@@ -0,0 +1,172 @@
+#[allow(unused)]
+use std::collections::HashMap;
+
+use async_trait::async_trait;
+#[allow(unused)]
+use datadog_protos::agent::{
+    GetFlareFilesRequest, GetFlareFilesResponse, GetStatusDetailsRequest, GetStatusDetailsResponse, RemoteAgent,
+};
+use memory_accounting::{MemoryBounds, MemoryBoundsBuilder};
+use saluki_config::GenericConfiguration;
+use saluki_core::components::{
+    destinations::{Destination, DestinationBuilder, DestinationContext},
+    ComponentContext,
+};
+use saluki_env::helpers::remote_agent::RemoteAgentClient;
+use saluki_error::GenericError;
+use saluki_event::DataType;
+use serde::Deserialize;
+use tracing::debug;
+
+const DEFAULT_ID: &str = "agent-data-plane";
+const DEFAULT_DISPLAY_NAME: &str = "Agent Data Plane";
+const DEFAULT_API_ENDPOINT: &str = "localhost:13580";
+
+/// TODO
+/// wrapped client inside an `Option` to fix Deserialization problems
+#[derive(Deserialize)]
+pub struct DatadogStatusFlareConfiguration {
+    #[serde(default = "default_id", rename = "remote_agent_id")]
+    id: String,
```

Comment:
We don't really want this to be configurable.

The remote agent ID should be derived from the data plane full name (normalized, though) and a randomly generated UUID. Something like:

```rust
let app_details = saluki_metadata::get_app_details();
let formatted_full_name = app_details
    .full_name()
    .replace(" ", "-")
    .replace("_", "-")
    .to_lowercase();

# You'll need to pull in the `uuid` crate for this.
let agent_id = format!("{}-{}", formatted_full_name, uuid::Uuid::now_v7());
```

We would make the field skipped for deserialization, and then generate this in `DatadogStatusFlareConfiguration::from_configuration` one time. The goal is that is should be generated with a consistent format but uniquely per each process... such that we know to expect the format to be `<name>-<uuid>` [^1] for ADP, but that the UUID changes when we restart ADP, so that we can always tie one UUID to one particular process instance.

[^1]: It _could_ be another format, but it should be consistent every time you restart ADP, basically.